lib/pages/chat/send_file_dialog/send_file_dialog.dart

```diff
@@ -0,0 +1,227 @@
+        },
+        (right) async {
+          if (right is GetServerConfigSuccess) {
+            final mUploadSize = right.serverConfig.mUploadSize;
+            if (mUploadSize != null) {
+              maxMediaSizeNotifier.value = mUploadSize.toDouble();
+              updateHaveErrorFilesNotifier();
+              loadThumbnailsForMedia(await getFilesNotError());
+              return;
+            }
+          }
+        },
+      );
+    });
+  }
+
+  void updateHaveErrorFilesNotifier() {
+    haveErrorFilesNotifier.value = filesNotifier.value.any(
+      (file) => file.size > maxMediaSizeNotifier.value,
+    );
+  }
+
+  Future<void> loadThumbnailsForMedia(List<MatrixFile> files) async {
+    if (widget.room == null) {
+      return;
+    }
+    generateThumbnailsMediaInteractor
+        .execute(
+      room: widget.room!,
+      files: files,
+      onConvertReadStreamToBytes: (oldFile, newFile) {
+        filesNotifier.update(oldFile, newFile);
+      },
+    )
+        .listen((event) {
+      event.fold(
+        (left) {
+          Logs().e("loadThumbnailsForMedia:: $left");
+        },
+        (right) {
+          if (right is GenerateThumbnailsMediaSuccess) {
+            thumbnails[right.file] = right.thumbnail;
+            filesNotifier.notify();
+          }
+        },
+      );
+    });
+  }
+
+  void requestFocusCaptions() {
+    captionsFocusNode.requestFocus();
+  }
+
+  void sendMediaWithCaption() {
+    if (widget.room == null) {
+      Logs().e("sendMediaWithCaption:: room is null");
+      Navigator.of(context).pop(SendMediaWithCaptionStatus.error);
+      return;
+    }
+    if (filesNotifier.value.isEmpty) {
+      return;
+    }
+    sendMediaOnWebWithCaptionInteractor.execute(
+      room: widget.room!,
+      media: filesNotifier.value.first,
+      caption: textEditingController.text,
+    );
+    Navigator.of(context).pop(SendMediaWithCaptionStatus.done);
+  }
+
+  Future<List<MatrixFile>> getFilesNotError() async {
+    return filesNotifier.value
+        .where((file) => !file.isFileHaveError(maxMediaSizeNotifier.value))
+        .toList();
+  }
+
+  Future<List<MatrixFile>> convertFilesToBytes(List<MatrixFile> files) async {
+    final results = await Future.wait(
+      files
+          .map(
+            (file) => file.convertReadStreamToBytes(),
+          )
+          .toList(),
+    );
+    for (int i = 0; i < files.length; i++) {
+      filesNotifier.update(files[i], results[i]);
+    }
+    return results;
+  }
+
+  void sendFilesWithCaption() async {
+    if (widget.room == null) {
+      Logs().e("sendFilesWithCaption:: room is null");
+      Navigator.of(context).pop(SendMediaWithCaptionStatus.error);
+      return;
+    }
+
+    sendFilesOnWebWithCaptionInteractor.execute(
+      room: widget.room!,
+      files: await getFilesNotError(),
+      caption: textEditingController.text,
+      thumbnails: thumbnails,
+    );
+    Navigator.of(context).pop(SendMediaWithCaptionStatus.done);
```

Comment:
IMO: After making a call, `SendMediaWithCaptionStatus` is done? I think this is not very clear, how can you be sure of success?

What happens if it fails?

Reply:
from send_file_dialog, it doesn't care sending file is successful or not, let the chat handle it